File: Semestr 3/Programowanie Obiektowe/Miniprojekt6/WyjatkiJava/main/Address.java
package main;
import java.util.Scanner;



public class Address
{
    Scanner sc = new Scanner(System.in);

    private String city, street, number;
    private boolean checkAddress(String c, String s, String n)
    {
        if(!c.matches("[a-zA-Z]+"))
        {
            return false;
        }

        if(!s.matches("[a-zA-Z]+"))
        {
            return false;
        }

        if(!n.matches("[0-9]+"))
        {
            return false;
        }
        return true;
    }


    Address(){}
    Address(String c, String s, String n)
    {
        while(true)
        {
            if(!checkAddress(c, s, n))
            {
                System.out.println("Niepoprawny adres, Wprowadz ponownie: ");
                c = sc.nextLine();
                s = sc.nextLine();
                n = sc.nextLine();
                continue;
            }
            this.city = c;
            this.street = s;
            this.number = n;
            break;
        }
    }


    public void setAddress(String c, String s, String n)
    {
        while(true)
        {
            if(!checkAddress(c, s, n))
            {
                System.out.println("Niepoprawny adres, Wprowadz ponownie: ");
                c = sc.nextLine();
                s = sc.nextLine();
                n = sc.nextLine();
                continue;
            }
            this.city = c;
            this.street = s;
            this.number = n;
            break;
        }
    }

    public void setCity(String c)
    {
        while(true)
        {
            if(!checkAddress(c, this.street, this.number))
            {
                System.out.println("Niepoprawne miasto, Wprowadz ponownie: ");
                c = sc.nextLine();
                continue;
            }
            this.city = c;
            break;
        }
    }

    public void setStreet(String s)
    {
        while(true)
        {
            if(!checkAddress(this.city, s, this.number))
            {
                System.out.println("Niepoprawna ulica, Wprowadz ponownie: ");
                s = sc.nextLine();
                continue;
            }
            this.street = s;
            break;
        }
    }

    public void setNumber(String n)
    {
        while(true)
        {
            if(!checkAddress(this.city, this.street, n))
            {
                System.out.println("Niepoprawny numer, Wprowadz ponownie: ");
                n = sc.nextLine();
                continue;
            }
            this.number = n;
            break;
        }
    }


    public Address getAddres(){ return this; }
    public String getCity(){ return this.city; }
    public String getStreet(){ return this.street; }
    public String getNumber(){ return this.number; }

    public String getAddressString()
    {
        return this.city + " " + this.street + " " + this.number;
    }

}

File: Semestr 3/Programowanie Obiektowe/Miniprojekt5/Address.cpp
#include "Address.h"

Address::Address():city(nullptr), street(nullptr), number(nullptr)
{
    city = new std::string;
    street = new std::string;
    number = new std::string;
}

Address::Address(std::string c, std::string s, std::string n)
{
    city = new std::string;
    street = new std::string;
    number = new std::string;

    while(true)
    {
        if(!checkAddress(c,s,n))
        {
            std::cout<<"Niepoprawny adres. Wprowadz ponownie: "<<std::endl;
            std::cin>>c>>s>>n;
            continue;
        }
        *city=c;
        *street=s;
        *number=n;
        break;
    }
}

Address::Address(const Address &a)
{
    city = new std::string;
    street = new std::string;
    number = new std::string; 
    
    *city=a.getCity();
    *street=a.getStreet();
    *number=a.getNumber();
}




bool Address::checkAddress(std::string c, std::string s, std::string n)
{
        for (int i = 0; i < c.size(); i++)
        {
            if (toupper(c[i]) < 'A' || toupper(c[i]) > 'Z')
            {
                return false;
            }
        }

        for (int i = 0; i < s.size(); i++)
        {
            if (toupper(s[i]) < 'A' || toupper(s[i]) > 'Z')
            {
                return false;
            }
        }

        for (int i = 0; i < n.size(); i++)
        {
            if (n[i] < '0' || n[i] > '9')
            {
                return false;
            }
        }
        return true;
}


void Address::setAddress(std::string c, std::string s, std::string n)
{
    while(true)
    {
        if(!checkAddress(c,s,n))
        {
            std::cout<<"Niepoprawny adres. Wprowadz ponownie: "<<std::endl;
            std::cin>>c>>s>>n;
            continue;
        }
        *city=c;
        *street=s;
        *number=n;
        break;
    }
}

void Address::setCity(std::string c)
{
    while(true)
    {
        if(!checkAddress(c,*street,*number))
        {
            std::cout<<"Niepoprawne miasto. Wprowadz ponownie: "<<std::endl;
            std::cin>>c;
            continue;
        }
        *city=c;
        break;
    }
}

void Address::setStreet(std::string s)
{
    while(true)
    {
        if(!checkAddress(*city,s,*number))
        {
            std::cout<<"Niepoprawna ulica. Wprowadz ponownie: "<<std::endl;
            std::cin>>s;
            continue;
        }
        *street=s;
        break;
    }
}

void Address::setNumber(std::string n)
{
    while(true)
    {
        if(!checkAddress(*city,*street,n))
        {
            std::cout<<"Niepoprawny numer. Wprowadz ponownie: "<<std::endl;
            std::cin>>n;
            continue;
        }
        *number=n;
        break;
    }
}



std::string Address::getCity() const
{
    return *city;
}

std::string Address::getStreet() const
{
    return *street;
}

std::string Address::getNumber() const
{
    return *number;
}

Address Address::getAddress() const
{
    return *this;
}

std::string Address::getAddressString() const
{
    return *city + " " + *street + " " + *number;
}